<launch>
    <arg name="namespace"                          default="drone1" />
    <arg name="activate_behavior_srv"              default="activate_behavior" />
    <arg name="deactivate_behavior_srv"            default="deactivate_behavior" />
    <arg name="check_activation_conditions_srv"    default="check_activation_conditions" />
    <arg name="activation_finished_topic"          default="behavior_activation_finished" />
    <arg name="estimated_pose_topic"               default="self_localization/pose" />
    <arg name="estimated_speed_topic"              default="self_localization/speed" />
    <arg name="motion_reference_speed_topic"       default="motion_reference/speed" />
    <arg name="motion_reference_pose_topic"        default="motion_reference/pose" />
    <arg name="motion_reference_path_topic"        default="motion_reference/path" />
    <arg name="controllers_topic"                  default="command/high_level" />
    <arg name="behavior_system"                    default="quadrotor_motion_with_pid_control" />
    <arg name="status_topic"                       default="status" />
    <arg name="set_control_mode_service_name"      default="set_control_mode" />
    


   <group ns="$(arg namespace)/$(arg behavior_system)">
        <param name="~namespace"                           value="$(arg namespace)"                          type="str" />
        <param name="~activate_behavior_srv"               value="$(arg activate_behavior_srv)"              type="str" />
        <param name="~deactivate_behavior_srv"             value="$(arg deactivate_behavior_srv)"            type="str" />
        <param name="~check_activation_conditions_srv"     value="$(arg check_activation_conditions_srv)"    type="str" />
        <param name="~estimated_pose_topic"                value="$(arg estimated_pose_topic)"               type="str" />
        <param name="~estimated_speed_topic"                value="$(arg estimated_speed_topic)"               type="str" />
        <param name="~controllers_topic"                   value="$(arg controllers_topic)"                  type="str" />
        <param name="~behavior_system"                     value="$(arg behavior_system)"                    type="str" />
        <param name="~status_topic"                          value="$(arg status_topic)" type="str" /> 
            
       <node pkg="nodelet" type="nodelet" args="manager" name="quadrotor_motion_with_pid_control_manager" output="screen" />
       <node pkg="nodelet" type="nodelet" args="load quadrotor_motion_with_pid_control/BehaviorKeepHoveringWithPidControl quadrotor_motion_with_pid_control_manager"
        name="behavior_keep_hovering_with_pid_control" output="screen"/>
       <node pkg="nodelet" type="nodelet" args="load quadrotor_motion_with_pid_control/BehaviorRotateWithPidControl quadrotor_motion_with_pid_control_manager"
        name="behavior_rotate_with_pid_control" output="screen"/>       
       <node pkg="nodelet" type="nodelet" args="load quadrotor_motion_with_pid_control/BehaviorKeepMovingWithPidControl quadrotor_motion_with_pid_control_manager"
        name="behavior_keep_moving_with_pid_control" output="screen"/>   
       <node pkg="nodelet" type="nodelet" args="load quadrotor_motion_with_pid_control/BehaviorFollowPath quadrotor_motion_with_pid_control_manager"
        name="behavior_follow_path" output="screen"/>                          
  </group>
</launch>
